kotlinx-coroutines-core/jvm/src/CoroutineExceptionHandlerImpl.kt

```diff
@@ -17,14 +18,29 @@ import kotlin.coroutines.*
  * We are explicitly using the `ServiceLoader.load(MyClass::class.java, MyClass::class.java.classLoader).iterator()`
  * form of the ServiceLoader call to enable R8 optimization when compiled on Android.
  */
-private val handlers: List<CoroutineExceptionHandler> = ServiceLoader.load(
+private val handlers: List<CoroutineExceptionHandler> by lazy {
+    ServiceLoader.load(
         CoroutineExceptionHandler::class.java,
         CoroutineExceptionHandler::class.java.classLoader
-).iterator().asSequence().toList()
+    ).iterator().asSequence().toList()
+}
 
 internal actual fun initializeDefaultExceptionHandlers() {
+    /**
+     * For Android, we do not want to allow arbitrary disks read (service loader JAR scanning)
+     * as it causes ANR, so we are postponing service loading until the actual exception (crash in case of Android).
+     * For other applications it's not the issue to read the JAR, but it is the issue
```

Comment:
```suggestion
     * For Android, we do not want to allow arbitrary disk reads (service loader JAR scanning)
     * as this would cause ANR, so we are postponing service loading until the actual exception (crash in case of Android).
     * For other applications, it's not an issue to read the JAR, but it is an issue
```